src/Lykke.Service.BlockchainWallets/Controllers/WalletsController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Lykke.Service.BlockchainWallets.Core.Services;
+using Lykke.Service.BlockchainWallets.Models;
+using Microsoft.AspNetCore.Mvc;
+
+
+namespace Lykke.Service.BlockchainWallets.Controllers
+{
+    [Route("api/wallets/{integrationLayerId}/{integrationLayerAssetId}")]
+    public class WalletsController : Controller
+    {
+        private readonly IBlockchainIntegrationService _blockchainIntegrationService;
+        private readonly IWalletService                _walletService;
+
+
+        public WalletsController(
+            IBlockchainIntegrationService blockchainIntegrationService,
+            IWalletService walletService)
+        {
+            _walletService     = walletService;
+            _blockchainIntegrationService = blockchainIntegrationService;
+        }
+
+
+        [HttpPost("by-client-ids/{clientId}")]
+        public async Task<IActionResult> CreateWallet([FromRoute] string integrationLayerId, [FromRoute] string integrationLayerAssetId, [FromRoute] Guid clientId)
+        {
+            if (!ValidateRequest(integrationLayerId, integrationLayerAssetId, clientId))
+            {
+                return BadRequest();
```

Comment:
Will be nice to see reasons of the bad request on the client side. You can use built-in validation attributes, check for ModelState.IsValid, and pass ModelState to the ErrorResponseFactory.Create()